ext/simdjson/src/bindings.cpp

```diff
@@ -0,0 +1,238 @@
+                add_property_zval_ex(&obj, field.key.data(), strlen(field.key.data()), &value);
+            }
+            v = obj;
+            break;
+        default:
+            break;
+    }
+    return v;
+}
+
+bool cplus_simdjson_is_valid(const char *json, size_t len)
+{
+    simdjson::dom::element doc;
+    auto error = parser.parse(json, len).get(doc);
+    if (error) {
+        return false;
+    }
+    return true;
+}
+
+void cplus_simdjson_parse(const char *json, size_t len, zval *return_value, unsigned char assoc, u_short depth)
+{
+    simdjson::dom::element doc;
+    auto error = build_parsed_json_cust(doc, json, len, true, depth);
+    if (error) {
+        zend_throw_error(php_json_exception_ce, simdjson::error_message(error), 0);
+        RETURN_THROWS();
+    }
+
+    if (assoc) {
+        *return_value = create_array(doc);
+    } else {
+        *return_value = create_object(doc);
+    }
+}
+
+void cplus_simdjson_key_value(
+	const char *json, size_t len, const char *key, zval *return_value, unsigned char assoc, u_short depth
+) {
+    simdjson::dom::element doc;
+    simdjson::dom::element element;
+    auto error = build_parsed_json_cust(doc, json, len, true, depth);
+    if (error) {
+        zend_throw_error(php_json_exception_ce, simdjson::error_message(error), 0);
+        RETURN_THROWS();
+    }
+
+    error = doc.at(key).get(element);
+
+    if (error) {
+        zend_throw_error(php_json_exception_ce, simdjson::error_message(error), 0);
+        RETURN_THROWS();
+    }
+
+    if (assoc) {
+        *return_value = create_array(element);
+    } else {
+        *return_value = create_object(element);
+    }
+}
+
+u_short cplus_simdjson_key_exists(const char *json, size_t len, const char *key, u_short depth)
+{
+    simdjson::dom::element doc;
+    auto error = build_parsed_json_cust(doc, json, len, true, depth);
+    if (error) {
+        return SIMDJSON_PARSE_KEY_NOEXISTS;
+    }
+    error = doc.at(key).error();
+    if (error) {
+        return SIMDJSON_PARSE_KEY_NOEXISTS;
+    }
+    return SIMDJSON_PARSE_KEY_EXISTS;
+}
+
+void cplus_simdjson_key_count(const char *json, size_t len, const char *key, zval *return_value, u_short depth)
+{
+    simdjson::dom::element doc;
+    simdjson::dom::element element;
+
+    auto error = build_parsed_json_cust(doc, json, len, true, depth);
+    if (error) {
+        zend_throw_error(php_json_exception_ce, simdjson::error_message(error), 0);
+        RETURN_THROWS();
+    }
+
+    error = doc.at(key).get(element);
+    if (error) {
+        zend_throw_error(php_json_exception_ce, simdjson::error_message(error), 0);
+        RETURN_THROWS();
+    }
+
+    zval v;
+    switch (element.type()) {
+        //ASCII sort
+        case simdjson::dom::element_type::ARRAY : ZVAL_LONG(&v, uint64_t(simdjson::dom::array(element).size()));
+            break;
+        case simdjson::dom::element_type::OBJECT : ZVAL_LONG(&v, uint64_t(simdjson::dom::object(element).size()));
```

Comment:
These would be clearer as `RETURN_LONG(uint64_t(simdjson::dom::object(element).size()));`

That macro returns from the function as well, making the break; redundant (it combines RETVAL_LONG and return;)